src/beamlime/workflow_protocols.py

```diff
@@ -67,4 +67,27 @@ def provide_stateless_workflow(workflow: Workflow) -> StatelessWorkflow:
     return stateless_workflows[workflow].load()
 
 
-dummy_workflow = DummyWorkflow()
+dummy_stateless_workflow = DummyStatelessWorkflow()
+
+
+@runtime_checkable
+class LiveWorkflow(Protocol):
+    def __init__(self, nexus_filename: Path) -> None:
+        """Initialize the workflow with all static information.
+
+        The only argument we need is the path to the nexus file or
+        a file object that workflow can read from,
+        since the file carrys all the static information.
+        """
+        ...
+
+    def __call__(
+        self, nxevent_data: dict[str, JSONGroup], nxlog: dict[str, JSONGroup]
+    ) -> WorkflowResult:
+        """Call the workflow and return the computed results that can be visualized.
+
+        ``nxevent_data`` should contain empty events even if no events was received
+        since the last call.
+        ``nxlog`` should only contain keys that we received since the last call.
```

Comment:
Can we use the standard docstring format? https://scipp.github.io/development/coding-conventions.html#docstrings

Reply:
@SimonHeybrock 
Done...!